### ISSUE TITLE
download Fail

### ISSUE THREAD

**Issue by shenlanchenwei** [hgouveia/node-downloader-helper#12]
When the file name of the downloaded file is too long, the file download will fail.

**Comment by hgouveia** [comment]
hello @shenlanchenwei , 

this could be  because of Windows path limitation 
https://docs.microsoft.com/en-us/windows/desktop/fileio/naming-a-file#maximum-path-length-limitation

>In the Windows API (with some exceptions discussed in the following paragraphs), the maximum length for a path is MAX_PATH, which is defined as 260 characters.

do you have any error message when this happens? if so could you copy and paste in here the output?

for the meantime i suggest to workaround, use the  `fileName` option to override the original name , but this will be a fixed name ,

or you could in a ugly way , modifying the function that takes care of the filename in the code 

https://github.com/hgouveia/node-downloader-helper/blob/master/src/index.js#L224
```js

const oldFunc = dl.__getFileNameFromHeaders;
dl.__getFileNameFromHeaders = function (headers) {
  const fileName = oldFunc(header);
  // handle the fileName, like cutting it 
  return myNewFileName;
}
```

Ideally the fix and what i will do , as suggested in this ticket https://github.com/hgouveia/node-downloader-helper/issues/10, is to allow the `fileName` option parameter to accept a Function, so the renaming could be handled by this custom function

**Comment by hgouveia** [comment]
Closed, added in version `1.0.11`
